src/cirrus/github_tools.py

```diff
@@ -218,6 +242,65 @@ def delete_branch(self, branch_name, remote=True):
         if remote:
             self.repo.git.push('origin', '--delete', branch_name)
 
+    def iter_github_branches(self):
+        """
+        iterate over branch names using the GH API.
+
+        Warning: This is subject to rate limiting
+        for repos with lots of branches
+
+        """
+        url = "https://api.github.com/repos/{org}/{repo}/branches".format(
+            org=self.config.organisation_name(),
+            repo=self.config.package_name()
+        )
+        params = {'per_page': 100}
+        resp = self.session.get(url, params=params)
+        resp.raise_for_status()
+        data = resp.json()
+        for row in data:
+            yield row['name']
+        next_page = resp.links['next']
+        while next_page:
+            resp = self.session.get(next_page['url'], params=params)
+            resp.raise_for_status()
+            data = resp.json()
+            for row in data:
+                yield row['name']
+            next_page = resp.links.get('next')
+            if next_page is None:
+                break
```

Comment:
Yeah, although ill change the while clause to while next_page is not None so that it is clear what happens to break the loop.